### PR TITLE
test: pin invalid timezone error on EncounterRecencyTransformer

### LINKED ISSUE #204: test: no test pins the invalid-timezone error that #202 restored in `EncounterRecencyTransformer`
### What's wrong

#202 (closing #201) deleted a `try`/`except` in `EncounterRecencyTransformer._parse_dates` that caught every exception from `.dt.tz_convert(self.timezone)` and retried with `.dt.tz_localize(self.timezone)`. The removal is correct: the same method parses with `utc=(self.timezone is not None)`, so whenever the fallback could run the series is already tz-aware and `tz_convert` cannot fail for a valid timezone. The only way the `except` ever fired was an **invalid** timezone name, and there it replaced the lookup error with a `TypeError` reading "Already tz-aware", which names the wrong problem entirely.

No test covers that. `tests/test_encounter_timezone.py` has five tests and all five pass a valid timezone or none at all; there is no assertion anywhere that an invalid timezone raises, and none about which exception comes out. So the behaviour #202 restored is unguarded: someone could reintroduce a bare `except` around that call, or swap the `utc=` argument back to a constant, and the whole suite would stay green while the misleading error came back.

This is the same shape as #200 and #157: a behaviour that only exists on a path no test walks.

### Where

| File | Locate with |
|---|---|
| `philanthropy/preprocessing/_encounter_recency.py` | `grep -n "tz_convert" philanthropy/preprocessing/_encounter_recency.py` |
| `tests/test_encounter_timezone.py` | `grep -n "def test_encounter_recency_localises" tests/test_encounter_timezone.py` |

### Expected behaviour, so it does not have to be guessed

An invalid timezone name must surface as a **lookup error naming the offending timezone**, not as a complaint about the data.

The concrete type differs by pandas major, so do not pin it directly. Both of these were run against `main` at the merge of #202:

| Environment | Exception | `str(e)` |
|---|---|---|
| pandas 2.3.3 (pytz present) | `pytz.exceptions.UnknownTimeZoneError` | `"'Not/AZone'"` |
| pandas 3.0.5 (resolves via `zoneinfo`) | `zoneinfo.ZoneInfoNotFoundError` | `"'No time zone found with key Not/AZone'"` |

`pyproject.toml` declares `pandas>=2.0` with no upper bound and the CI matrix spans py3.9 to py3.13 plus a lowest-direct floors job, so **both rows really happen in CI**. Pinning `pytz.exceptions.UnknownTimeZoneError` would fail on the pandas 3 jobs, and `pytz` is not a declared dependency of this project, so importing it in a test is not safe either.

**What to pin instead:** both exceptions subclass `KeyError`, and the pre-#202 bug raised `TypeError("Already tz-aware")`, which does not. So `KeyError` plus the timezone name in the message discriminates the regression on every supported pandas, with no `pytz` import:

```python
with pytest.raises(KeyError, match="Not/AZone"):
    ...
```

Do not match on `Exception` or `ValueError`. Both would pass against the bug, which is the thing this test exists to catch.

### What to change

No source change. This is test-only.

1. Read `test_encounter_recency_localises_to_a_non_utc_timezone` in `tests/test_encounter_timezone.py` for the fixture and assertion style used there; reuse its two-row `X`.
2. Add a test that constructs the transformer with a timezone name that does not exist, calls `fit_transform`, and asserts with `pytest.raises(KeyError, match=...)` that the offending name appears in the message.
3. Add an assertion (in the same test or a second one) that the message does **not** contain "Already tz-aware", so the test fails loudly if the masking fallback is ever reintroduced rather than silently passing on the wrong exception.

`tests/conftest.py` pins `TZ=UTC`, so this is reproducible off a CI runner with no extra setup.

Please keep the PR to the test. If you think `_parse_dates` should validate `timezone` earlier, or re-raise something friendlier, say so in the PR description and leave it for a separate issue; `__init__` doing no validation is a deliberate scikit-learn convention here (see AGENTS.md) and changing it is a larger decision than this issue.

### Verifying

```
python -m pip install -e ".[dev]"
python -m pytest tests/test_encounter_timezone.py -q
make ci
```

Every PR also needs a `## [Unreleased]` entry in `CHANGELOG.md` and your own name added to `CONTRIBUTORS.md`, in the same PR. CI enforces both.

Follow-up to #201 / #202.

### PR DESCRIPTION
**Pull request by HarshRajSinghania**
Closes #204.

## Summary
Adds a regression test so an invalid `timezone=` on `EncounterRecencyTransformer` raises `KeyError` that names the zone. The pre-#202 fallback turned a bad zone into `TypeError("Already tz-aware")`; this test would fail if that path returned.

No production code change. `__init__` still does not validate `timezone` (sklearn convention noted in the issue).

## Motivation
#204: `tests/test_encounter_timezone.py` only used valid timezones or none. The behaviour restored in #202 was unguarded.

## Implementation
- Reuse the two-row fixture from `test_encounter_recency_localises_to_a_non_utc_timezone`.
- Construct the transformer with `timezone="Not/AZone"` and call `fit_transform`.
- Assert `pytest.raises(KeyError, match="Not/AZone")` so both pandas 2 (pytz) and pandas 3 (zoneinfo) are covered without importing pytz.
- Assert the message does not contain `Already tz-aware`.
- `CHANGELOG.md` `[Unreleased]` entry and `CONTRIBUTORS.md` line, as required by the project.

## Testing
Intended local commands from the issue:
```
python -m pip install -e ".[dev]"
python -m pytest tests/test_encounter_timezone.py -q
make ci
```
This environment could not install scikit-learn (PyPI 502), so the new test was not executed here. CI on this PR should run the file and the full matrix.

### CODE REVIEW

**Comment by shivamlalakiya** [comment]
Closing this as a duplicate: #204 was claimed by @Kaap10 on 2026-09-12 and confirmed on the issue the same day, and #221 has now merged as dba8cd5.

That is the claim-before-PR rule from #187 working as intended, and it is on me that it is easy to miss: the claim is a comment on the issue rather than an assignee, so the issue looks unclaimed from the list view. Checking the issue thread before starting is the only reliable way right now.

Nothing wrong with the test itself. If you want another, #50, #53 and #55 are all open with zero comments, so no claim to collide with:

- #50 `PropensityScorer.predict_proba` single-class branch
- #53 cli `_score_array` plain `predict_proba` fallback branch
- #55 `FiscalYearGroupedSplitter.split` no-prior-training-data skip branch

Thanks for #216 and for this one.
